### PR TITLE
Buffs carps again

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Normal and Mega carps have 10~ 20~ more HP
Mega carps lower damage add on is now 4 to 10 rather then 2 to 10
The Nukie carp now can regen passively 
## Why It's Good For The Game

The nukie carp is a ghost role that will likely take a few hits and then die, or sit on the shuttle/space well the team dies/wins the game mode. Having it stuck in a "man if only they would heal me" mode kinda sucks and makes them more usefull - Note it takes 11 mins to fully regen from 2 hp to full

As for carp hp buffs, they go down in 2-3 hits with a spear/tool box and are not that much of a threat to anyone in space. 

Mega carps are buffed a bit more so that they are less of a hit it /4 to 5!/ times and have them bit a less squishy for being a massive mob that can mess you up if you dont prepare for the fight

## Changelog
:cl:
add: Carps have evolved to take more damage
/:cl:

### CODE REVIEW

**Comment by Arturlang** [comment]
Sounds good, carp really needed a buff anyways.

**Comment by BlueWildrose** [comment]
An indirect buff to the dehydrated space carp traitor item too. Nice.

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 1]
```diff
@@ -104,4 +104,14 @@
...
+/mob/living/simple_animal/hostile/carp/cayenne/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(.)
+		regen_cooldown = world.time + REGENERATION_DELAY
+
+/mob/living/simple_animal/hostile/carp/cayenne/Life()
+	. = ..()
+	if(regen_cooldown < world.time)
+		heal_overall_damage(8)
+
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]: Move this procedure to the base space carp type and add a new variable to toggle the health regeneration for Cayenne and mega carps.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 2]
```diff
@@ -29,6 +33,20 @@
 	attack_sound = 'sound/weapons/bite.ogg'
 	speak_emote = list("gnashes")
 
+	//Some carps heal over time
+
+/mob/living/simple_animal/hostile/carp/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(regen != TRUE)
+		return
+	else
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]: Replace these three lines with
```
	if(regen)
```

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 3]
```diff
@@ -29,6 +33,20 @@
...
+		regen_cooldown = world.time + REGENERATION_DELAY
+
+/mob/living/simple_animal/hostile/carp/Life()
+	. = ..()
+	if(regen_cooldown < world.time)
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]:
```suggestion
	if(regen && regen_cooldown < world.time)
```

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 4]
```diff
@@ -17,10 +17,14 @@
...
 	speed = 0
-	maxHealth = 25
-	health = 25
+	maxHealth = 35
+	health = 35
 	spacewalk = TRUE
 
+	var/regen = FALSE //Can it heal over time or not?
+	var/regen_cooldown = 0 //Used for how long it takes before a healing will take place default in 60 seconds
+	var/heal_amout = 0 //How much is healed pre regen cooldown
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]: You could merge `regen` and `heal_amount` into an unique var.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**